xmtp_mls/src/builder.rs

```diff
@@ -136,7 +136,7 @@ where
                 parameter: "api_client",
             })?;
         let network = self.network;
-        let store = self.store.take().unwrap_or_default();
+        let store = self.store.take().unwrap();
```

Comment:
I'm not sure having a default here is really helpful outside of tests. If an actual caller isn't specifying a store their application is almost certainly misconfigured and panicking makes sense.

Reply:
As far as panicking goes, I think we should let the caller decide whether to panic or not and just stick to errors. We have more control over the error context and can return more helpful results than with panics. Additionally it allows any external crate or service using the library to recover if they choose to

Reply:
If someone were to accidentally use the default (having the data stored in memory or in a tempfile), the result is total data loss.

Reply:
I don't disagree that we can't continue, but a couple lines above this we return an error instead of panicking on missing an api_client, which is a similar case to this. i think we should choose a way here and be consistent throughout

At the very least, we should use `.expect` here to inform the client library what went wrong, and document it on the `ClientBuilder` fn that it panics under some circumstances

Reply:
If we don't have good defaults for these, it might be better to fit them into the constructor, and then we avoid this issue altogether b/c it will be impossible to enter an invalid state with the builder

Reply:
That's a good point. I've updated to be consistent with above.